src/objects/api/fields.py

```diff
@@ -78,3 +85,12 @@ def get_url(self, obj, view_name, request, format):
         lookup_value = getattr(obj.object, "uuid")
         kwargs = {self.lookup_url_kwarg: lookup_value}
         return self.reverse(view_name, kwargs=kwargs, request=request, format=format)
+
+
+class CachedObjectUrlField(CachedHyperlinkedIdentityField):
+    lookup_field = "uuid"
+
+    def get_url(self, obj, view_name, request, format):
+        if hasattr(obj, "pk") and obj.pk in (None, ""):
+            return None
```

Comment:
This is implemented in the `CacheMixin` from `commonground-api-common` as well (https://github.com/maykinmedia/commonground-api-common/blob/main/vng_api_common/serializers.py#L313-L314), so I don't think this subclass is needed, you could import `CachedHyperlinkedIdentityField` and do `url = CachedHyperlinkedIdentityField(view_name="...", lookup_field="uuid")`